apps/frontend/src/pages/projects/components/project-popover/project-popover.tsx

```diff
@@ -0,0 +1,85 @@
+import { MenuItem, Modal, Popover } from "~/libs/components/components.js";
+import { DataStatus } from "~/libs/enums/enums.js";
+import {
+	useAppDispatch,
+	useAppSelector,
+	useCallback,
+	useEffect,
+	useModal,
+	usePopover,
+} from "~/libs/hooks/hooks.js";
+import {
+	actions as projectActions,
+	type ProjectGetAllItemResponseDto,
+	type ProjectPatchRequestDto,
+} from "~/modules/projects/projects.js";
+
+import { ProjectUpdateForm } from "../components.js";
+import styles from "./styles.module.css";
+
+type Properties = {
+	children: React.ReactNode;
+	project: ProjectGetAllItemResponseDto;
+};
+
+const ProjectPopover = ({ children, project }: Properties): JSX.Element => {
+	const dispatch = useAppDispatch();
+
+	const { projectPatchStatus } = useAppSelector(({ projects }) => projects);
+
+	const { isPopoverOpened, onPopoverClose, onPopoverOpen } = usePopover();
+	const { isModalOpened, onModalClose, onModalOpen } = useModal();
+
+	const handleEditClick = useCallback((): void => {
+		onPopoverClose();
+		onModalOpen();
+	}, [onPopoverClose, onModalOpen]);
+
+	useEffect(() => {
+		if (projectPatchStatus === DataStatus.FULFILLED) {
+			onModalClose();
+		}
+	}, [projectPatchStatus, onModalClose]);
+
+	const handleProjectEditSubmit = useCallback(
+		(id: number, payload: ProjectPatchRequestDto) => {
+			void dispatch(projectActions.patch({ id, payload }));
+		},
+		[dispatch],
+	);
+
+	return (
+		<>
+			<Popover
+				content={
+					<div className={styles["project-menu-popover"]}>
+						<div className={styles["project-menu-items"]}>
+							<MenuItem
+								iconName="pencil"
+								label="Edit"
+								onClick={handleEditClick}
+							/>
+						</div>
+					</div>
+				}
+				isPopoverOpened={isPopoverOpened}
+				onPopoverClose={onPopoverClose}
+				onPopoverOpen={onPopoverOpen}
+			>
+				{children}
```

Comment:
1. keep the trigger here as a part of component because it's a specific component, not a common one
2. Use IconButton as a trigger, not usual button. That's why we added usePopover hook, to be able to pass onClick to trigger, outside of popover.
3. Related to the 2 point, we need to remove button wrapper for trigger element inside popover. And then we will need to wrap avatar inside button from outside
```suggestion
			<IconButton onClick={onPopoverOpen} />
```